### PR TITLE
SAK-43741: ORACLE - DB blocked by a lot of Update lesson_builder_items calls

### PR DESCRIPTION
**Pull request by jesusmmp**

### CODE REVIEW

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ReorderProducer.java [hunk 1]
```diff
@@ -189,7 +189,10 @@ public void fillComponents(UIContainer tofill, ViewParameters params, ComponentC
 					    UIOutput.make(row, "text-snippet", messageLocator.getMessage("simplepage.resources-snippet"));
 				    }
 				    else {
-					    String text = FormattedText.convertFormattedTextToPlaintext(i.getHtml());
+					    String text = i.getHtml();
+					    if (text == null)
+						    text = "";
```
Comment (ern) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ReorderProducer.java:194]: put this on the same line or use braces

**Comment by ern** [comment]
Have you run this production, is changing these to null cause any NPE's in other parts of the code?

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ReorderProducer.java (hunk 1 above)
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ReorderProducer.java:194]: Add some braces here?

File: lessonbuilder/hbm/src/java/org/sakaiproject/lessonbuildertool/SimplePageItemImpl.java [hunk 2]
```diff
@@ -179,9 +179,7 @@ public SimplePageItemImpl(long pageId, int sequence, int type, String sakaiId, S
     // are too long. The main case this will happen is if we generate a name, which we do for URLs
     // and uploaded files.
 	private String maxlength(String s, int maxlen) {
-	    if (s == null)
-		s = "";  // oracle turns "" into null
-	    int len = s.length();
+	    int len = s==null?0:s.length();
```
Comment (adrianfish) [inline lessonbuilder/hbm/src/java/org/sakaiproject/lessonbuildertool/SimplePageItemImpl.java:182]: Stretch this out a bit? s == null ? 0 : s.length(). It's clearer, I think.

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ReorderProducer.java (hunk 1 above)
Comment (jesusmmp) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ReorderProducer.java:194]: Done

Comment (jesusmmp) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ReorderProducer.java:194]: Done

File: lessonbuilder/hbm/src/java/org/sakaiproject/lessonbuildertool/SimplePageItemImpl.java (hunk 2 above)
Comment (jesusmmp) [reply lessonbuilder/hbm/src/java/org/sakaiproject/lessonbuildertool/SimplePageItemImpl.java]: Done

**Comment by jesusmmp** [comment]
> Have you run this production, is changing these to null cause any NPE's in other parts of the code?

We have this patch in our production instance 12.4 since 06/05/2020. 
@juanjmerono could add more information if needed.

**Pull request merged**